spring/src/main/java/org/axonframework/spring/config/SpringLifecycleRegistry.java

```diff
@@ -81,7 +81,7 @@ public LifecycleRegistry onShutdown(int phase, @Nonnull LifecycleHandler shutdow
                 phase,
                 () -> shutdownHandler.run(beanFactory.getBean(AxonConfiguration.class))
         );
-        beanFactory.registerSingleton(getBeanName("shutdown") + uniqueId.getAndIncrement(), springShutdownHandler);
+        beanFactory.registerSingleton(getBeanName("shutdown"), springShutdownHandler);
```

Comment:
This is a small additional fix I did combined with the above. I assume that's alright with the reviewers :pray: